mne/bem.py

```diff
@@ -1108,12 +1107,112 @@ def make_watershed_bem(subject, subjects_dir=None, overwrite=False,
 
     logger.info('Created %s\n\nComplete.' % (fname_head,))
 
 
+def convert_surface(fname_in, fname_out, T1_mgz, useLAS=False):
+    """Convert surface.
+
+    Parameters
+    ----------
+    fname_in : str
+        Path to surface in FreeSurfer surf format.
+    fname_out : str
+        File to write.
+    T1_mgz : str
+        Full path to T1.mgz file.
+    useLAS :
+        Flip x axis.
+
+    Returns
+    -------
+    surf : dict
+        Converted surface.
+    """
+    from .source_space import _get_mgz_header
+    from .surface import _read_surface_geom, write_surface
+    header = _get_mgz_header(T1_mgz)
+    surf_in = _read_surface_geom(fname_in)
+    surf_in['coord_frame'] = FIFF.FIFFV_COORD_MRI
+    ex, ey, ez, _ = header['ras2vox'][:3].T
+    rot = np.zeros((3, 3))
+
+    ex /= np.linalg.norm(ex)
+    ey /= np.linalg.norm(ey)
+    ez /= np.linalg.norm(ez)
+    right = ant = sup = ex
+
+    if (abs(ey[0]) > abs(right[0])):
+        right = ey
+    if (abs(ez[0]) > abs(right[0])):
+        right = ez
+    if (abs(ey[1]) > abs(ant[1])):
+        ant = ey
+    if (abs(ez[1]) > abs(ant[1])):
+        ant = ez
+    if (abs(ey[2]) > abs(sup[2])):
+        sup = ey
+    if (abs(ez[2]) > abs(sup[2])):
+        sup = ez
+
+    if np.all(right == ant) or np.all(right == sup) or np.all(ant == sup):
+        raise ValueError("Cannot decide the RAS directions.")
+
+    if right[0] < 0:
+        right *= -1.
+    if ant[1] < 0:
+        ant *= -1.
+    if sup[2] < 0:
+        sup *= -1.
+    for j in range(3):
+        if useLAS:
+            rot[j][0] = -right[j]
+        else:
+            rot[j][0] = right[j]
+        rot[j][1] = ant[j]
+        rot[j][2] = sup[j]
+
+    corners = np.zeros((8, 3), dtype=float)
+    corners[1][0] = header['dims'][0] - 1.0
+    corners[2][0] = header['dims'][0] - 1.0
+    corners[2][1] = header['dims'][1] - 1.0
+    corners[3][1] = header['dims'][1] - 1.0
+    for j in range(4, 8):
+        corners[j][0] = corners[j - 4][0]
+        corners[j][1] = corners[j - 4][1]
+        corners[j][2] = header['dims'][2] - 1.0
+
+    r0 = np.zeros(3, dtype=float)
+    for j in range(8):
+        corners[j] = apply_trans(header['ras2vox'], corners[j], move=True)
+        r0 += corners[j]
+    r0 /= 8.0
+
+    dir = (-1.0, -1.0, -1.0)
+    maxdot = -1.
+    LPI_RPI = None
+    for j in range(8):
+        corners[j] -= r0
+        thisdot = np.dot(corners[j], dir) / (np.linalg.norm(corners[j]) *
+                                             np.linalg.norm(dir))
+        if thisdot > maxdot:
+            LPI_RPI = corners[j]
+            maxdot = thisdot
+    if LPI_RPI is None:
+        raise ValueError("Could not determine the corner")
+
+    move = LPI_RPI + r0
+    trans = np.vstack((np.vstack((rot, move)).T, np.zeros(4)))
+    trans[3][3] = 1.
+    trans = Transform(FIFF.FIFFV_COORD_UNKNOWN, FIFF.FIFFV_COORD_MRI, trans)
+    surf_out = transform_surface_to(surf_in, FIFF.FIFFV_COORD_MRI, trans)
```

Comment:
How can this work when you have `surf_in['coord_frame'] = FIFF.FIFFV_COORD_MRI` above? Does this line even do anything? It should either do nothing, or it should throw an error about the transform not going to the correct coordinate frame. If it doesn't, I think we have a bug in `transform_surface_to` checks, since during this call the surface already appears to be in `MRI` coords.